packages/web-components/src/discount-card/discount-card.ts

```diff
@@ -0,0 +1,247 @@
+import { property } from "lit/decorators.js";
+import { classMap } from "lit/directives/class-map.js";
+import logger from "../utils/logger.ts";
+import { Slots } from "./constants.ts";
+import styles from "./discount-card.style.ts";
+
+/**
+ * @summary A component for displaying discount cards with various appearances
+ *
+ * @tag tapsi-discount-card
+ *
+ * @slot [action] - The slot for actions associated with the component, typically a collection of `tapsi-button` components.
+ * @slot [thumbnail] - The slot for custom image.
+ * @slot [header-icon] - The slot for header icon
+ */
+export class DiscountCard extends LitElement {
+  /** @internal */
+  public static override readonly styles = [styles];
+
+  /**
+   * The header title of the discount card.
+   *
+   * @prop {string} headerTitle
+   * @attr {string} header-title
+   * @default ""
+   */
+  @property({ attribute: "header-title" })
+  public headerTitle = "";
+
+  /**
+   * The variant of the discount card.
+   *
+   * @prop {"clay" | "whisper" | "azure" | "flame" | "grayscale" | "none"} variant
+   * @attr {"clay" | "whisper" | "azure" | "flame" | "grayscale" | "none"} variant
+   * @default "none"
+   */
+  @property({ reflect: true })
+  public variant:
+    | "clay"
+    | "whisper"
+    | "azure"
+    | "flame"
+    | "grayscale"
+    | "none" = "none";
+
+  /**
+   * The title to use for the discount-card.
+   * Represents the title of the discount-card.
+   *
+   * @prop {string} title
+   * @attr {string} title
+   * @default ""
+   */
+  @property()
+  public override title = "";
+
+  /**
+   * The description of the discount card.
+   *
+   * @prop {string} description
+   * @attr {string} description
+   * @default ""
+   */
+  @property()
+  public description = "";
+
+  /**
+   * The name/text to display in the badge.
+   *
+   * @prop {string} badgeName
+   * @attr {string} badge-name
+   * @default ""
+   */
+  @property({ attribute: "badge-name" })
+  public badgeName = "";
+
+  /**
+   * The label for the expiry date of the discount.
+   *
+   * @prop {string} expiryDateLabel
+   * @attr {string} expiry-date-label
+   * @default ""
+   */
+  @property({ attribute: "expiry-date-label" })
+  public expiryDateLabel = "";
+
+  /**
+   * Check if discount is expiring and if expiring is true then turn expiryDateLabel into red color
+   *
+   * @prop {boolean} expiring
+   * @attr {string} expiring
+    @default false
+   */
+  @property({ type: Boolean })
+  public expiring: boolean = false;
+
+  /**
+   * Check for required props and emit warnings
+   */
+  private checkRequiredProps() {
+    const isTransparent = this.variant === "none";
+
+    // Check if header icon slot is filled
+    const hasHeaderIcon = !!this.querySelector(`[slot="${Slots.HEADER_ICON}"]`);
+
+    if (!isTransparent && !this.headerTitle) {
+      logger(
+        "headerTitle is required when variant is not none",
+        "discount-card",
+        "warning",
+      );
```

Comment:
We usually wrap variables inside backticks in logs.

```suggestion
      logger(
        "`headerTitle` is required when `variant` is not `none`",
        "discount-card",
        "warning",
      );
```